mllib/src/test/scala/org/apache/spark/ml/evaluation/ClusteringEvaluatorSuite.scala

```diff
@@ -0,0 +1,225 @@
+      Row(Vectors.dense(7.1, 3.0, 5.9, 2.1), 2),
+      Row(Vectors.dense(6.3, 2.9, 5.6, 1.8), 2),
+      Row(Vectors.dense(6.5, 3.0, 5.8, 2.2), 2),
+      Row(Vectors.dense(7.6, 3.0, 6.6, 2.1), 2),
+      Row(Vectors.dense(4.9, 2.5, 4.5, 1.7), 2),
+      Row(Vectors.dense(7.3, 2.9, 6.3, 1.8), 2),
+      Row(Vectors.dense(6.7, 2.5, 5.8, 1.8), 2),
+      Row(Vectors.dense(7.2, 3.6, 6.1, 2.5), 2),
+      Row(Vectors.dense(6.5, 3.2, 5.1, 2.0), 2),
+      Row(Vectors.dense(6.4, 2.7, 5.3, 1.9), 2),
+      Row(Vectors.dense(6.8, 3.0, 5.5, 2.1), 2),
+      Row(Vectors.dense(5.7, 2.5, 5.0, 2.0), 2),
+      Row(Vectors.dense(5.8, 2.8, 5.1, 2.4), 2),
+      Row(Vectors.dense(6.4, 3.2, 5.3, 2.3), 2),
+      Row(Vectors.dense(6.5, 3.0, 5.5, 1.8), 2),
+      Row(Vectors.dense(7.7, 3.8, 6.7, 2.2), 2),
+      Row(Vectors.dense(7.7, 2.6, 6.9, 2.3), 2),
+      Row(Vectors.dense(6.0, 2.2, 5.0, 1.5), 2),
+      Row(Vectors.dense(6.9, 3.2, 5.7, 2.3), 2),
+      Row(Vectors.dense(5.6, 2.8, 4.9, 2.0), 2),
+      Row(Vectors.dense(7.7, 2.8, 6.7, 2.0), 2),
+      Row(Vectors.dense(6.3, 2.7, 4.9, 1.8), 2),
+      Row(Vectors.dense(6.7, 3.3, 5.7, 2.1), 2),
+      Row(Vectors.dense(7.2, 3.2, 6.0, 1.8), 2),
+      Row(Vectors.dense(6.2, 2.8, 4.8, 1.8), 2),
+      Row(Vectors.dense(6.1, 3.0, 4.9, 1.8), 2),
+      Row(Vectors.dense(6.4, 2.8, 5.6, 2.1), 2),
+      Row(Vectors.dense(7.2, 3.0, 5.8, 1.6), 2),
+      Row(Vectors.dense(7.4, 2.8, 6.1, 1.9), 2),
+      Row(Vectors.dense(7.9, 3.8, 6.4, 2.0), 2),
+      Row(Vectors.dense(6.4, 2.8, 5.6, 2.2), 2),
+      Row(Vectors.dense(6.3, 2.8, 5.1, 1.5), 2),
+      Row(Vectors.dense(6.1, 2.6, 5.6, 1.4), 2),
+      Row(Vectors.dense(7.7, 3.0, 6.1, 2.3), 2),
+      Row(Vectors.dense(6.3, 3.4, 5.6, 2.4), 2),
+      Row(Vectors.dense(6.4, 3.1, 5.5, 1.8), 2),
+      Row(Vectors.dense(6.0, 3.0, 4.8, 1.8), 2),
+      Row(Vectors.dense(6.9, 3.1, 5.4, 2.1), 2),
+      Row(Vectors.dense(6.7, 3.1, 5.6, 2.4), 2),
+      Row(Vectors.dense(6.9, 3.1, 5.1, 2.3), 2),
+      Row(Vectors.dense(5.8, 2.7, 5.1, 1.9), 2),
+      Row(Vectors.dense(6.8, 3.2, 5.9, 2.3), 2),
+      Row(Vectors.dense(6.7, 3.3, 5.7, 2.5), 2),
+      Row(Vectors.dense(6.7, 3.0, 5.2, 2.3), 2),
+      Row(Vectors.dense(6.3, 2.5, 5.0, 1.9), 2),
+      Row(Vectors.dense(6.5, 3.0, 5.2, 2.0), 2),
+      Row(Vectors.dense(6.2, 3.4, 5.4, 2.3), 2),
+      Row(Vectors.dense(5.9, 3.0, 5.1, 1.8), 2))
+
+  val dsStruct = StructType( Seq(
+    StructField("point", new VectorUDT, nullable = false),
+    StructField("label", IntegerType, nullable = false)
+  ))
+
+  test("params") {
+    ParamsSuite.checkParams(new ClusteringEvaluator)
+  }
+
+  test("read/write") {
+    val evaluator = new ClusteringEvaluator()
+      .setPredictionCol("myPrediction")
+      .setFeaturesCol("myLabel")
+      .setMetricName("squaredSilhouette")
+    testDefaultReadWrite(evaluator)
+  }
+
+  /*
+  Use the following python code to load the data and evaluate it using scikit-learn package.
```

Comment:
you should add the expected output of your python code, refer to [FPGrowthSuite.scala
](https://github.com/apache/spark/blob/39e2bad6a866d27c3ca594d15e574a1da3ee84cc/mllib/src/test/scala/org/apache/spark/mllib/fpm/FPGrowthSuite.scala#L71), and mind the indent